### PR TITLE
feat: Move scraper to queue

### PR DESCRIPTION
**Pull request by mogery**

### CODE REVIEW

**Comment by nickscamara** [comment]
- [ ] fix cancel

**Comment by nickscamara** [comment]
- [ ] Job Prioritization

**Comment by nickscamara** [comment]
- [ ] Fix biling

**Comment by nickscamara** [comment]
- [ ] Activity Logs and Usage Page
- [ ] Partial data maintain order

**Comment by nickscamara** [comment]
- [ ] @calebpeffer check if posthog is okay

**Comment by nickscamara** [comment]
- [ ] Webhook fixes

**Comment by nickscamara** [comment]
- [ ] @nickscamara's Load Testing

**Comment by calebpeffer** [comment]
> * [ ]  @calebpeffer check if posthog is okay

I don't have much context, so it's hard for me to do. I can't tell which events are coming from staging and which are coming from prod. It would be better if someone with a staging setup sent some particular job-logged calls and then searched for them. @nickscamara, it's probably you since you can access Posthog and the staging environment.

**Comment by nickscamara** [comment]
Food for thought: how do we prioritize /scrape and /crawl for request 0? Basically if a user (user1) is scraping 1000/min, how do we ensure that user2, that will /scrape will get a low response time on their request (both on the processing side(queue side) and on the API side (app))

**Comment by nickscamara** [comment]
- [ ] Higher priority on request 0

**Comment by nickscamara** [comment]
- [ ] Address app elevated response times when sending multiple /scrape per min

**Comment by mogery** [comment]
heading out for the day. next steps on this:
 - [ ] fix activity logs/usage page/SQL RPCs (should we just create crawl firecrawl_jobs with no data just metadata? and ignore the scrapes with crawl_id != NULL)
 - [ ] make sure posthog data is sane (we can put a dummy crawl on posthog alongside the webhook call code)
 - [ ] fix billing category (not sure where this is in the code)

File: apps/api/src/scraper/WebScraper/single_url.ts [hunk 1]
```diff
@@ -340,7 +340,7 @@ export async function scrapSingleUrl(
         Logger.debug(`⛏️ ${scraper}: Successfully scraped ${urlToScrap} with text length >= 100, breaking`);
         break;
       }
-      if (pageStatusCode && pageStatusCode == 404) {
+      if (pageStatusCode && (pageStatusCode == 404 || pageStatusCode == 500)) {
         Logger.debug(`⛏️ ${scraper}: Successfully scraped ${urlToScrap} with status code 404, breaking`);
```
Comment (nickscamara) [inline apps/api/src/scraper/WebScraper/single_url.ts:346]: Change the log debug status code

File: apps/api/src/scraper/WebScraper/utils/utils.ts
Comment (nickscamara) [inline apps/api/src/scraper/WebScraper/utils/utils.ts:1]: How to handle errors in `new URL`

**Comment by nickscamara** [comment]
- [ ] Test Self host

**Review by nickscamara — COMMENTED** [review]
Looking really good! @rafaelsideguide and I just did a deeper review on it

**Comment by rafaelsideguide** [comment]
> * [x]  Test Self Host w/ `use db false and fire-engine false` and docker setup

Tested with both local and docker. Working great!

**Comment by nickscamara** [comment]
- [ ] Pass job priority to fire-engine

**Comment by nickscamara** [comment]
- [X] Usage RPCs updated

**Review by nickscamara — APPROVED** [review]
LGTM!

**Pull request merged**